addons/l10n_in_edi/models/account_edi_format.py

```diff
@@ -345,6 +360,94 @@ def _get_l10n_in_edi_line_details(self, index, line, line_tax_details, sign):
+                    'CesAmt': self._l10n_in_round_value(discount_line_vals['CesAmt'] + other_line_vals['CesAmt']),
+                    'CesNonAdvlAmt': self._l10n_in_round_value(discount_line_vals['CesNonAdvlAmt'] + other_line_vals['CesNonAdvlAmt']),
+                    'StateCesAmt': self._l10n_in_round_value(discount_line_vals['StateCesAmt'] + other_line_vals['StateCesAmt']),
+                    'StateCesNonAdvlAmt': self._l10n_in_round_value(discount_line_vals['StateCesNonAdvlAmt'] + other_line_vals['StateCesNonAdvlAmt']),
+                    'OthChrg': self._l10n_in_round_value(discount_line_vals['OthChrg'] + other_line_vals['OthChrg']),
+                    'TotItemVal': self._l10n_in_round_value(discount_line_vals['TotItemVal'] + other_line_vals['TotItemVal']),
+                })
+                other_line_vals.update({
+                    'Discount': self._l10n_in_round_value(other_line_vals['Discount'] + discount_to_allow),
+                    'AssAmt': 0.00,
+                    'IgstAmt': 0.00,
+                    'CgstAmt': 0.00,
+                    'SgstAmt': 0.00,
+                    'CesAmt': 0.00,
+                    'CesNonAdvlAmt': 0.00,
+                    'StateCesAmt': 0.00,
+                    'StateCesNonAdvlAmt': 0.00,
+                    'OthChrg': 0.00,
+                    'TotItemVal': 0.00,
+                })
+                return False
+            other_line_vals.update({
+                'Discount': self._l10n_in_round_value(other_line_vals['Discount'] + discount),
+                'AssAmt': self._l10n_in_round_value(other_line_vals['AssAmt'] + discount_line_vals['AssAmt']),
+                'IgstAmt': self._l10n_in_round_value(other_line_vals['IgstAmt'] + discount_line_vals['IgstAmt']),
+                'CgstAmt': self._l10n_in_round_value(other_line_vals['CgstAmt'] + discount_line_vals['CgstAmt']),
+                'SgstAmt': self._l10n_in_round_value(other_line_vals['SgstAmt'] + discount_line_vals['SgstAmt']),
+                'CesAmt': self._l10n_in_round_value(other_line_vals['CesAmt'] + discount_line_vals['CesAmt']),
+                'CesNonAdvlAmt': self._l10n_in_round_value(other_line_vals['CesNonAdvlAmt'] + discount_line_vals['CesNonAdvlAmt']),
+                'StateCesAmt': self._l10n_in_round_value(other_line_vals['StateCesAmt'] + discount_line_vals['StateCesAmt']),
+                'StateCesNonAdvlAmt': self._l10n_in_round_value(other_line_vals['StateCesNonAdvlAmt'] + discount_line_vals['StateCesNonAdvlAmt']),
+                'OthChrg': self._l10n_in_round_value(other_line_vals['OthChrg'] + discount_line_vals['OthChrg']),
+                'TotItemVal': self._l10n_in_round_value(other_line_vals['TotItemVal'] + discount_line_vals['TotItemVal']),
+            })
+            return True
+
+        discount_lines = []
+        # reverse lines to remove from last
+        for discount_line in json_payload['ItemList'][::-1]:
```

Comment:
```suggestion
        for discount_line in json_payload['ItemList'].copy(): #to be sure to not skip in the loop
```
Otherwise people will wonder each time why the reverse is done.

Reply:
okay